tests/VolleyManagement.Specs/TeamsContext/ManageTeamPlayersSteps.cs

```diff
@@ -1,87 +1,161 @@
+                    .SingleOrDefault(p => p.FirstName == names.FirstName && p.LastName == names.LastName);
+                _playersToRemove.Add(new PlayerId(playerEntities.Id));
+            }
         }
-        
-        [Given(@"Jane Doe is a team captain")]
-        public void GivenJaneDoeIsATeamCaptain()
+
+        [Given(@"(.*) is a team captain")]
+        public void GivenJaneDoeIsATeamCaptain(string playerName)
         {
-            ScenarioContext.Current.Pending();
+            var names = SpecsHelper.SplitFullNameToFirstLastNames(playerName);
+
+            using (var ctx = TestDbAdapter.Context)
+            {
+                var playerEntity = ctx.Players
+                    .SingleOrDefault(p => p.Id == _captainId.Id);
+
+                if (playerEntity == null)
+                {
+                    return;
+                }
+
+                playerEntity.FirstName = names.FirstName;
+                playerEntity.LastName = names.LastName;
+                ctx.SaveChanges();
+            }
         }
-        
+
         [When(@"I execute AddPlayersToTeam")]
         public void WhenIExecuteAddPlayersToTeam()
         {
-            ScenarioContext.Current.Pending();
+            _teamService.AddPlayers(_teamId, _playerToAdd);
         }
-        
+
         [When(@"I execute RemovePlayersFromTeam")]
         public void WhenIExecuteRemovePlayersFromTeam()
         {
-            ScenarioContext.Current.Pending();
+            try
+            {
+                _teamService.RemovePlayers(_teamId, _playersToRemove);
+            }
+            catch (ArgumentException exc)
+            {
+                _exception = exc;
+            }
         }
-        
+
         [Then(@"players are added")]
         public void ThenPlayersAreAdded()
         {
-            ScenarioContext.Current.Pending();
+            using (var ctx = TestDbAdapter.Context)
+            {
+                IEnumerable<PlayerEntity> playerEntities = ctx.Teams.First(team => team.Id == _teamId.Id).Players;
+                var entityRosterIds = playerEntities.Select(e => e.Id);
+                var playersId = _playerToAdd.Select(e => e.Id);
+                entityRosterIds.Should().Contain(playersId, "Players should be added");
+            }
         }
-        
+
         [Then(@"players are removed")]
         public void ThenPlayersAreRemoved()
         {
-            ScenarioContext.Current.Pending();
+            using (var ctx = TestDbAdapter.Context)
+            {
+                IEnumerable<PlayerEntity> playerEntities = ctx.Teams.First(team => team.Id == _teamId.Id).Players;
+                var entityRosterIds = playerEntities.Select(e => e.Id);
+                if (_playersToRemove != null)
+                {
+                    entityRosterIds.Should().NotContain(_playersToRemove, "Players should be removed");
+                }
+                _exception.Should().BeNull("Should not thrown exception");
```

Comment:
Can this be not null?

Reply:
no.

Reply:
Do we need it then?

Reply:
we don't need it, becouse it's positive scenario